enterprise/coderd/coderd.go

```diff
@@ -61,10 +62,26 @@ func New(ctx context.Context, options *Options) (_ *API, err error) {
 	}
 
 	ctx, cancelFunc := context.WithCancel(ctx)
-	api := &API{
-		ctx:    ctx,
-		cancel: cancelFunc,
 
+	if options.ExternalTokenEncryption != nil {
+		cryptDB, err := dbcrypt.New(ctx, options.Database, options.ExternalTokenEncryption...)
+		if err != nil {
+			cancelFunc()
+			// If we fail to initialize the database, it's likely that the
+			// database is encrypted with an unknown external token encryption key.
+			// This is a fatal error.
+			var derr *dbcrypt.DecryptFailedError
+			if xerrors.As(err, &derr) {
+				panic(`Coder has shut down to prevent data corruption: your configured database is encrypted with an unknown external token encryption key. Please check your configuration and try again.`)
```

Comment:
is `panic` intended or would a regular error be enough?

BTW you may add an information something informing about the fact that encryption is performed within/without transaction enabled.

Reply:
The `panic` results in a marginally more useful output to the user than the error.
The INFO log about database encryption is a good call 👍

Reply:
I don't think this should be a panic because the user will be confused by the stacktrace. Should keep this as a returned error instead, and you should include detials about which key is missing and details about how to recover from this situation (link to docs)